src/Microsoft.AspNet.Hosting/WebApplicationBuilder.cs

```diff
@@ -32,111 +33,33 @@ public class WebHostBuilder
         private StartupMethods _startup;
         private Type _startupType;
         private string _startupAssemblyName;
-        private readonly bool _captureStartupErrors;
 
         // Only one of these should be set
         private string _serverFactoryLocation;
         private IServerFactory _serverFactory;
         private IServer _server;
+        private string _webRoot;
 
-        public WebHostBuilder()
-            : this(config: new ConfigurationBuilder().Build())
+        public WebApplicationBuilder()
         {
-        }
-
-        public WebHostBuilder(IConfiguration config)
-            : this(config: config, captureStartupErrors: false)
-        {
-        }
-
-        public WebHostBuilder(IConfiguration config, bool captureStartupErrors)
-        {
-            if (config == null)
-            {
-                throw new ArgumentNullException(nameof(config));
-            }
-
             _hostingEnvironment = new HostingEnvironment();
             _loggerFactory = new LoggerFactory();
-            _config = config;
-            _options = new WebHostOptions(config);
-            _captureStartupErrors = captureStartupErrors;
+            _config = WebApplicationConfiguration.GetDefault();
```

Comment:
Let _config start as null and then lazy init it in Build. Otherwise you will usually build config twice since UseConfiguration will usually be called.